finagle-core/src/test/scala/com/twitter/finagle/filter/RequestMeterFilterTest.scala

```diff
@@ -0,0 +1,44 @@
+package com.twitter.finagle.filter
+
+import com.twitter.concurrent.AsyncMeter
+import com.twitter.conversions.time._
+import com.twitter.finagle.util.DefaultTimer
+import com.twitter.finagle.{Failure, Service}
+import com.twitter.util.{Await, Future}
+import org.junit.runner.RunWith
+import org.scalatest.FunSuite
+import org.scalatest.junit.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class RequestMeterFilterTest extends FunSuite {
+
+  val timer = DefaultTimer.twitter
+  val meter = AsyncMeter.newMeter(1, 1 second, 1)(timer)
+
+  test("return service execution after getting a permit") {
+    val echoSvc = new Service[Int, Int] {
+      def apply(req: Int) = Future(req)
+    }
+    val svc = new RequestMeterFilter(meter) andThen echoSvc
+    assert(Await.result(svc(2)) == 2)
+  }
+
+  test("mark dropped requests as rejected") {
+    val neverSvc = new Service[Int, Int] {
+      def apply(req: Int) = Future.never
+    }
+    val svc = new RequestMeterFilter(meter, 2) andThen neverSvc
+    val f = intercept[Failure] { Await.result(svc(1)) }
+    assert(f.isFlagged(Failure.Restartable))
+  }
+
+  test("service failures are not wrapped as rejected") {
+    val exc = new Exception("app exc")
```

Comment:
The idea of this test is to make the meter throw an exception because I'm trying to acquire more than 1 permit within a second, and check the exception type, etc. I've changed it to advance after the first call to meter. I've tried advancing a couple hundred millisecs but it still hangs. Only if I advance 1 sec it stops hanging, but it doesn't behave the way I expect, as it doesn't throw any exception. What am I missing? Thanks... 

```scala
val timer = new MockTimer
Time.withCurrentTimeFrozen { ctl =>
  val meter = AsyncMeter.newMeter(1, 1 second, 1)(timer)
  meter.await(1)

  ctl.advance(1 second)
  timer.tick()

  Await.result(meter.await(1))
  println("done!")
}
```